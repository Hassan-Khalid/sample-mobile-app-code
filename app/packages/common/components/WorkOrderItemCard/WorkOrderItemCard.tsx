import React, { useMemo } from 'react';
import { View, Text as RNText, TouchableOpacity } from 'react-native';
import styles from './styles/WorkOrderItemCard.styles';
import { COMMON_LOGOS } from '../../assets/Images';
import Text from "../Text/Text"
import Image from "../CustomImage/Image"
import { Avatar } from '@rneui/themed';
import UIUtil from 'common/util/UIUtil';
import { AnyType, Status } from 'common/declarations/types';
import {
  capitalizeFirstLetter,
  generateInitials,
  getWorkOrderCreatedAt
} from '../../../oem/features/WorkOrder/WorkOrderListing/helper/workOrderHelper';
import AppColors from 'common/assets/colors/AppColors';
import { WorkOrderListTypeObject } from 'common/declarations/global';
import WorkOrderTypeIcons from 'common/components/WorkOrderTypeIcon/WorkOrderTypeIcons';
import NavigationService from 'common/services/NavigationService';
import SCREENS from 'common/constants/Screens';
type workOrderItemType = {
  item:WorkOrderListTypeObject
  statuses:AnyType[] | undefined,
  onPress:(item)=>void
}

const WorkOrderItemCard = (props:workOrderItemType) => {
  const WorkOrderIDView = ()=>{
    let iconWrapperStyles = useMemo(() => UIUtil.generateStatusBackgroundColor(props.item?.ticketType?.color), [props.item?.ticketType?.color]);
  return(
    <View style={[styles.topViewContainer,styles.rowContainer]}>
      <View style={[styles.rowContainer]}>
        <View style={{...styles.ticketTypeIconContainer,...{backgroundColor:iconWrapperStyles}}}>
          <WorkOrderTypeIcons
            icon={props.item.ticketType?.icon}
            strokeColor={props.item.ticketType?.color}
            size={16}
          />
        </View>
        <View style={{...styles.rowContainer,...{marginHorizontal:10, }}}>
          <RNText>
          <Text style={styles.lightText}>{props.item.ticketId}</Text>
          <Text> • </Text>
          <Text style={styles.lightText}>{getWorkOrderCreatedAt(props.item.createdAt)}</Text>
          </RNText>
        </View>
      </View>
      <View style={[styles.rowContainer]}>
        <Text style={styles.fileText}>{props.item.procedures?.length}</Text>
        <Image style={styles.fileIcon}  source={COMMON_LOGOS.IC_PROCEDURE} resizeMode={'contain'}/>
      </View>
    </View>
  )
}
const WorkOrderDetailView = () => {
    const {machine,facility} = props.item;
    return(
       <View style={[styles.centerViewContainer]}>
         <View style={{}}>
           <Text numberOfLines={2} ellipsizeMode="tail" style={styles.workOrderHeading}>{props.item.title}</Text>
         </View>
         <View style={{...styles.rowContainer}}>
             <Text numberOfLines={1} ellipsizeMode="tail" style={styles.workOrderSubHeading}>{facility?.name}</Text>
             <Text  style={styles.workOrderSubHeading}>{' • '}</Text>
             <Text numberOfLines={1}  ellipsizeMode="tail" style={styles.workOrderCompanyName}>{machine?.name}</Text>
           <Text  style={styles.workOrderSubHeading}>{' • '}</Text>
           <Text numberOfLines={1}  ellipsizeMode="tail" style={styles.workOrderMachineName}>{machine?.serialNumber}</Text>
         </View>
       </View>
    )}
 const UserImage:React.FC = () =>{
   const {assignee} = props.item
    if(assignee?.name){
      return (
        <>
          <Avatar
            size={26}
            rounded
            titleStyle={{color:AppColors.BLUE}}
            containerStyle={{ backgroundColor: AppColors.BLUE_LIGHT }}
            title={generateInitials(assignee?.name)}
          />
          <Text numberOfLines={1}  ellipsizeMode="tail" style={styles.titleStyle}>{capitalizeFirstLetter(assignee?.name)}</Text>
        </>
    )}
    else{
    return(
      <View style={styles.rowContainer}>
        <Image source={COMMON_LOGOS.IC_USER} resizeMode={'contain'} style={styles.avatar}/>
        <Text numberOfLines={1} style={styles.titleStyle}>{'tickets.ticketDetails.notAssigned'}</Text>
      </View>
    )}
  }
const WorkOrderStatusView = () => {
    const {status} = props.item
    const statusObject:Status | undefined  = UIUtil.getStatus(props.statuses,status)
    return(
      <View style={[styles.bottomViewContainer,styles.rowContainer]}>
        <View style={styles.rowContainer}>
          <UserImage/>
        </View>
        <View style={{...styles.statusView,backgroundColor:UIUtil.generateStatusBackgroundColor(statusObject?.color)}}>
          <Text style={{...styles.statusText,color:statusObject?.color}}>{statusObject?.label.toUpperCase()}</Text>
        </View>
      </View>
    )
  }
  return (
    <View >
      <TouchableOpacity onPress={()=>NavigationService.navigate(SCREENS.WORK_ORDER_DETAIL,{id:props.item._id})} style={styles.container}>
      <View style={styles.innerContainer}>
        <WorkOrderIDView/>
        <WorkOrderDetailView/>
        <WorkOrderStatusView/>
      </View>
      </TouchableOpacity>
    </View>
  );
};

export default WorkOrderItemCard;
