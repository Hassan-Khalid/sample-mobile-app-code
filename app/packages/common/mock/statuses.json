[
  {
    "__typename":"Statuses",
    "_id":"64d373c44c4ed3ebaccaa2d5",
    "color":"#EE0064",
    "label":"Open"
  },
  {
    "__typename":"Statuses",
    "_id":"64d373c44c4ed3ebaccaa2d6",
    "color":"#00A5BC",
    "label":"CALLBACK"
  },
  {
    "__typename":"Statuses",
    "_id":"64d373c44c4ed3ebaccaa2d7",
    "color":"#00b09b",
    "label":"VISIT PLANNED"
  },
  {
    "__typename":"Statuses",
    "_id":"65d61c3416735b0a70af8c14",
    "color":"#6974fb",
    "label":"READY FOR DEV"
  },
  {
    "__typename":"Statuses",
    "_id":"64eb3bb6a6dde661d6f671fe",
    "color":"#172b4d",
    "label":"SURVEY DONE"
  },
  {
    "__typename":"Statuses",
    "_id":"6532162d14fb774162ace44e",
    "color":"#9e4b65",
    "label":"PARTS ORDERED"
  },
  {
    "__typename":"Statuses",
    "_id":"64d373c44c4ed3ebaccaa2d8",
    "color":"#8940e5",
    "label":"ON HOLD"
  },
  {
    "__typename":"Statuses",
    "_id":"64db4f75b7a453ddccdea83b",
    "color":"#FF7A00",
    "label":"TROUBLESHOOTING"
  },
  {
    "__typename":"Statuses",
    "_id":"65f166bd6b9d3febc9ef63bd",
    "color":"#ee0064",
    "label":"EN-ROUTE"
  },
  {
    "__typename":"Statuses",
    "_id":"64d373c44c4ed3ebaccaa2da",
    "color":"#00B09B",
    "label":"Closed"
  }
]