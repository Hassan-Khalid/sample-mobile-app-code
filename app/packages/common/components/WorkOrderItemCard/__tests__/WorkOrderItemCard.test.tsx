import React from 'react';
import { render } from '@testing-library/react-native';
import WorkOrderItemCard from '../WorkOrderItemCard';
import mockWorkOrder from "common/mock/workOrder.json"
import statuses from "common/mock/statuses.json"


jest.mock('../../../../oem/features/WorkOrder/WorkOrderListing/helper/workOrderhelper', () => ({
  getWorkOrderCreatedAt: jest.fn().mockReturnValue('2024-03-19'), // Mocked return value
  capitalizeFirstLetter: jest.fn().mockImplementation((str) => {
    if (typeof str !== 'string' || str.trim().length === 0) {
      throw new Error('Input must be a non-empty string.');
    }
    return str.replace(/\b\w/g, char => char.toUpperCase());
  }),
  generateInitials: jest.fn().mockImplementation((name) => {
    const words = name.split(' ');
    return words.map(part => part.charAt(0)).join('').toUpperCase();
  }),
}));

describe('WorkOrderItemCard', () => {
  
  
  it('renders correctly', () => {
    // exercise
    const wrapper = render(<WorkOrderItemCard item={mockWorkOrder[0]} statuses = {statuses}/>);
    
    // post-conditions
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
  
  it('renders correctly with no Assignee', () => {
    // exercise
    const wrapper = render(<WorkOrderItemCard item={mockWorkOrder[1]} statuses = {statuses}/>);
    
    // post-conditions
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
  
});
