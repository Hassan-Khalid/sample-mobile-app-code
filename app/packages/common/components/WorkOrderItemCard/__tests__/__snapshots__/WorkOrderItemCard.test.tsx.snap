// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkOrderItemCard renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": {
        "dynamic": {
          "dark": "#1b1b1b",
          "highContrastDark": undefined,
          "highContrastLight": undefined,
          "light": "#FFFFFF",
        },
      },
      "borderColor": {
        "dynamic": {
          "dark": "#2a2d32",
          "highContrastDark": undefined,
          "highContrastLight": undefined,
          "light": "#E8EAED",
        },
      },
      "borderRadius": 10,
      "borderWidth": 1,
      "height": 180,
      "marginHorizontal": 30,
      "width": 715,
    }
  }
>
  <View
    style={
      {
        "flex": 1,
        "justifyContent": "space-between",
        "margin": 15,
      }
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "justifyContent": "space-between",
          },
          {
            "alignItems": "center",
            "flexDirection": "row",
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "flexDirection": "row",
            },
          ]
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#CDD1FE",
              "borderRadius": 15,
              "height": 30,
              "justifyContent": "center",
              "width": 30,
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={20}
            bbWidth={20}
            fill="none"
            focusable={false}
            height={20}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                {
                  "flex": 0,
                  "height": 20,
                  "width": 20,
                },
              ]
            }
            vbHeight={20}
            vbWidth={20}
            width={20}
            xml="<svg
          width="20"
          height="20"
          viewBox="0 0 20 20"
          fill="none"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M2.5 16.25V5C2.5 4.83424 2.56585 4.67527 2.68306 4.55806C2.80027 4.44085 2.95924 4.375 3.125 4.375H7.28906C7.4242 4.37556 7.55562 4.41936 7.66406 4.5L9.83594 6.125C9.94438 6.20564 10.0758 6.24944 10.2109 6.25H15.625C15.7908 6.25 15.9497 6.31585 16.0669 6.43306C16.1842 6.55027 16.25 6.70924 16.25 6.875V8.75"
            stroke='#0517f8'
            strokeWidth="1.4"
            strokeLinecap="round"
            strokeLinejoin="round"
          />
          <path
            d="M2.5 16.25L4.85938 9.17969C4.9005 9.05467 4.98004 8.94583 5.08666 8.86868C5.19327 8.79152 5.32152 8.74999 5.45313 8.75H17.8828C17.9816 8.74999 18.0791 8.77342 18.1671 8.81836C18.2551 8.86331 18.3312 8.92848 18.3892 9.00854C18.4471 9.0886 18.4852 9.18126 18.5004 9.27891C18.5156 9.37657 18.5074 9.47643 18.4766 9.57031L16.25 16.25H2.5Z"
            stroke='#0517f8'
            strokeWidth="1.4"
            strokeLinecap="round"
            strokeLinejoin="round"
          />
        </svg>"
            xmlns="http://www.w3.org/2000/svg"
          >
            <RNSVGGroup
              fill={null}
              propList={
                [
                  "fill",
                ]
              }
            >
              <RNSVGPath
                d="M2.5 16.25V5C2.5 4.83424 2.56585 4.67527 2.68306 4.55806C2.80027 4.44085 2.95924 4.375 3.125 4.375H7.28906C7.4242 4.37556 7.55562 4.41936 7.66406 4.5L9.83594 6.125C9.94438 6.20564 10.0758 6.24944 10.2109 6.25H15.625C15.7908 6.25 15.9497 6.31585 16.0669 6.43306C16.1842 6.55027 16.25 6.70924 16.25 6.875V8.75"
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
                propList={
                  [
                    "stroke",
                    "strokeWidth",
                    "strokeLinecap",
                    "strokeLinejoin",
                  ]
                }
                stroke={
                  {
                    "payload": 4278523896,
                    "type": 0,
                  }
                }
                strokeLinecap={1}
                strokeLinejoin={1}
                strokeWidth={1.4}
              />
              <RNSVGPath
                d="M2.5 16.25L4.85938 9.17969C4.9005 9.05467 4.98004 8.94583 5.08666 8.86868C5.19327 8.79152 5.32152 8.74999 5.45313 8.75H17.8828C17.9816 8.74999 18.0791 8.77342 18.1671 8.81836C18.2551 8.86331 18.3312 8.92848 18.3892 9.00854C18.4471 9.0886 18.4852 9.18126 18.5004 9.27891C18.5156 9.37657 18.5074 9.47643 18.4766 9.57031L16.25 16.25H2.5Z"
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
                propList={
                  [
                    "stroke",
                    "strokeWidth",
                    "strokeLinecap",
                    "strokeLinejoin",
                  ]
                }
                stroke={
                  {
                    "payload": 4278523896,
                    "type": 0,
                  }
                }
                strokeLinecap={1}
                strokeLinejoin={1}
                strokeWidth={1.4}
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "marginHorizontal": 10,
            }
          }
        >
          <Text>
            <Text
              style={
                {
                  "color": {
                    "dynamic": {
                      "dark": "#77889d",
                      "highContrastDark": undefined,
                      "highContrastLight": undefined,
                      "light": "#77889d",
                    },
                  },
                }
              }
            >
              HGE2K5VS2R
            </Text>
            <Text
              style={
                {
                  "color": {
                    "dynamic": {
                      "dark": "#172B4D",
                      "highContrastDark": undefined,
                      "highContrastLight": undefined,
                      "light": "#172B4D",
                    },
                  },
                }
              }
            >
               â€¢ 
            </Text>
            <Text
              style={
                {
                  "color": {
                    "dynamic": {
                      "dark": "#77889d",
                      "highContrastDark": undefined,
                      "highContrastLight": undefined,
                      "light": "#77889d",
                    },
                  },
                }
              }
            >
              2024-03-19
            </Text>
          </Text>
        </View>
      </View>
      <View
        style={
          [
            {
              "alignItems": "center",
              "flexDirection": "row",
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": {
                "dynamic": {
                  "dark": "#748094",
                  "highContrastDark": undefined,
                  "highContrastLight": undefined,
                  "light": "#8798ad",
                },
              },
              "fontSize": 14,
              "marginHorizontal": 5,
            }
          }
        >
          4
        </Text>
        <View
          style={
            [
              {
                "overflow": "hidden",
              },
              {
                "height": 22,
                "width": 22,
              },
            ]
          }
        >
          <FastImageView
            accessibilityLabel="fast-image"
            defaultSource={null}
            resizeMode="contain"
            style={
              {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
            testID="fast-image"
          />
        </View>
      </View>
    </View>
    <View
      style={
        [
          {
            "flex": 1,
            "justifyContent": "space-evenly",
          },
        ]
      }
    >
      <View
        style={{}}
      >
        <Text
          ellipsizeMode="tail"
          numberOfLines={2}
          style={
            {
              "color": {
                "dynamic": {
                  "dark": "#172B4D",
                  "highContrastDark": undefined,
                  "highContrastLight": undefined,
                  "light": "#172B4D",
                },
              },
              "flexWrap": "wrap",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Testing 13.02.24
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <Text
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            {
              "color": {
                "dynamic": {
                  "dark": "#77889d",
                  "highContrastDark": undefined,
                  "highContrastLight": undefined,
                  "light": "#77889d",
                },
              },
              "flexShrink": 1,
              "fontSize": 14,
              "fontWeight": "400",
            }
          }
        >
          Argentina Customer
        </Text>
        <Text
          style={
            {
              "color": {
                "dynamic": {
                  "dark": "#77889d",
                  "highContrastDark": undefined,
                  "highContrastLight": undefined,
                  "light": "#77889d",
                },
              },
              "flexShrink": 1,
              "fontSize": 14,
              "fontWeight": "400",
            }
          }
        >
           â€¢ 
        </Text>
        <Text
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            {
              "color": {
                "dynamic": {
                  "dark": "#77889d",
                  "highContrastDark": undefined,
                  "highContrastLight": undefined,
                  "light": "#77889d",
                },
              },
              "flexShrink": 1,
              "flexWrap": "wrap",
              "fontSize": 14,
              "fontWeight": "400",
            }
          }
        >
          Argentina Machine
        </Text>
        <Text
          style={
            {
              "color": {
                "dynamic": {
                  "dark": "#77889d",
                  "highContrastDark": undefined,
                  "highContrastLight": undefined,
                  "light": "#77889d",
                },
              },
              "flexShrink": 1,
              "fontSize": 14,
              "fontWeight": "400",
            }
          }
        >
           â€¢ 
        </Text>
        <Text
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            {
              "color": {
                "dynamic": {
                  "dark": "#77889d",
                  "highContrastDark": undefined,
                  "highContrastLight": undefined,
                  "light": "#77889d",
                },
              },
              "fontSize": 14,
              "fontWeight": "400",
            }
          }
        >
          A-machine
        </Text>
      </View>
    </View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "justifyContent": "space-between",
          },
          {
            "alignItems": "center",
            "flexDirection": "row",
          },
        ]
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": {
                "dynamic": {
                  "dark": "#E6E8FE",
                  "highContrastDark": undefined,
                  "highContrastLight": undefined,
                  "light": "#E6E8FE",
                },
              },
              "borderRadius": 13,
              "display": "flex",
              "flexDirection": "column",
              "height": 26,
              "justifyContent": "center",
              "width": 26,
            }
          }
          theme={
            {
              "colors": {
                "background": "#ffffff",
                "black": "#242424",
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": {
                  "android": {
                    "error": "#f44336",
                    "grey": "rgba(0, 0, 0, 0.54)",
                    "primary": "#2196f3",
                    "searchBg": "#dcdce1",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "default": {
                    "error": "#ff3b30",
                    "grey": "#7d7d7d",
                    "primary": "#007aff",
                    "searchBg": "#dcdce1",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                  "ios": {
                    "error": "#ff3b30",
                    "grey": "#7d7d7d",
                    "primary": "#007aff",
                    "searchBg": "#dcdce1",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                  "web": {
                    "error": "#ff190c",
                    "grey": "#393e42",
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#ca71eb",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#ad1457",
                "success": "#52c41a",
                "warning": "#faad14",
                "white": "#ffffff",
              },
              "spacing": {
                "lg": 12,
                "md": 8,
                "sm": 4,
                "xl": 24,
                "xs": 2,
              },
            }
          }
        >
          <Text
            style={
              {
                "backgroundColor": "transparent",
                "color": {
                  "dynamic": {
                    "dark": "#64b4fa",
                    "highContrastDark": undefined,
                    "highContrastLight": undefined,
                    "light": "#0000EE",
                  },
                },
                "fontSize": 13,
                "textAlign": "center",
                "zIndex": 1,
              }
            }
          >
            H
          </Text>
        </View>
        <Text
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            {
              "fontSize": 14,
              "marginHorizontal": 10,
              "textAlign": "center",
            }
          }
        >
          Haider
        </Text>
      </View>
      <View
        style={
          {
            "backgroundColor": "#CCEFEB",
            "borderRadius": 10,
            "padding": 8,
          }
        }
      >
        <Text
          style={
            {
              "color": "#00b09b",
              "fontSize": 13,
            }
          }
        >
          VISIT PLANNED
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`WorkOrderItemCard renders correctly with no Assignee 1`] = `
<View
  style={
    {
      "backgroundColor": {
        "dynamic": {
          "dark": "#1b1b1b",
          "highContrastDark": undefined,
          "highContrastLight": undefined,
          "light": "#FFFFFF",
        },
      },
      "borderColor": {
        "dynamic": {
          "dark": "#2a2d32",
          "highContrastDark": undefined,
          "highContrastLight": undefined,
          "light": "#E8EAED",
        },
      },
      "borderRadius": 10,
      "borderWidth": 1,
      "height": 180,
      "marginHorizontal": 30,
      "width": 715,
    }
  }
>
  <View
    style={
      {
        "flex": 1,
        "justifyContent": "space-between",
        "margin": 15,
      }
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "justifyContent": "space-between",
          },
          {
            "alignItems": "center",
            "flexDirection": "row",
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "flexDirection": "row",
            },
          ]
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#CDD1FE",
              "borderRadius": 15,
              "height": 30,
              "justifyContent": "center",
              "width": 30,
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={20}
            bbWidth={20}
            fill="none"
            focusable={false}
            height={20}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                {
                  "flex": 0,
                  "height": 20,
                  "width": 20,
                },
              ]
            }
            vbHeight={20}
            vbWidth={20}
            width={20}
            xml="<svg
          width="20"
          height="20"
          viewBox="0 0 20 20"
          fill="none"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M2.5 16.25V5C2.5 4.83424 2.56585 4.67527 2.68306 4.55806C2.80027 4.44085 2.95924 4.375 3.125 4.375H7.28906C7.4242 4.37556 7.55562 4.41936 7.66406 4.5L9.83594 6.125C9.94438 6.20564 10.0758 6.24944 10.2109 6.25H15.625C15.7908 6.25 15.9497 6.31585 16.0669 6.43306C16.1842 6.55027 16.25 6.70924 16.25 6.875V8.75"
            stroke='#0517f8'
            strokeWidth="1.4"
            strokeLinecap="round"
            strokeLinejoin="round"
          />
          <path
            d="M2.5 16.25L4.85938 9.17969C4.9005 9.05467 4.98004 8.94583 5.08666 8.86868C5.19327 8.79152 5.32152 8.74999 5.45313 8.75H17.8828C17.9816 8.74999 18.0791 8.77342 18.1671 8.81836C18.2551 8.86331 18.3312 8.92848 18.3892 9.00854C18.4471 9.0886 18.4852 9.18126 18.5004 9.27891C18.5156 9.37657 18.5074 9.47643 18.4766 9.57031L16.25 16.25H2.5Z"
            stroke='#0517f8'
            strokeWidth="1.4"
            strokeLinecap="round"
            strokeLinejoin="round"
          />
        </svg>"
            xmlns="http://www.w3.org/2000/svg"
          >
            <RNSVGGroup
              fill={null}
              propList={
                [
                  "fill",
                ]
              }
            >
              <RNSVGPath
                d="M2.5 16.25V5C2.5 4.83424 2.56585 4.67527 2.68306 4.55806C2.80027 4.44085 2.95924 4.375 3.125 4.375H7.28906C7.4242 4.37556 7.55562 4.41936 7.66406 4.5L9.83594 6.125C9.94438 6.20564 10.0758 6.24944 10.2109 6.25H15.625C15.7908 6.25 15.9497 6.31585 16.0669 6.43306C16.1842 6.55027 16.25 6.70924 16.25 6.875V8.75"
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
                propList={
                  [
                    "stroke",
                    "strokeWidth",
                    "strokeLinecap",
                    "strokeLinejoin",
                  ]
                }
                stroke={
                  {
                    "payload": 4278523896,
                    "type": 0,
                  }
                }
                strokeLinecap={1}
                strokeLinejoin={1}
                strokeWidth={1.4}
              />
              <RNSVGPath
                d="M2.5 16.25L4.85938 9.17969C4.9005 9.05467 4.98004 8.94583 5.08666 8.86868C5.19327 8.79152 5.32152 8.74999 5.45313 8.75H17.8828C17.9816 8.74999 18.0791 8.77342 18.1671 8.81836C18.2551 8.86331 18.3312 8.92848 18.3892 9.00854C18.4471 9.0886 18.4852 9.18126 18.5004 9.27891C18.5156 9.37657 18.5074 9.47643 18.4766 9.57031L16.25 16.25H2.5Z"
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
                propList={
                  [
                    "stroke",
                    "strokeWidth",
                    "strokeLinecap",
                    "strokeLinejoin",
                  ]
                }
                stroke={
                  {
                    "payload": 4278523896,
                    "type": 0,
                  }
                }
                strokeLinecap={1}
                strokeLinejoin={1}
                strokeWidth={1.4}
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "marginHorizontal": 10,
            }
          }
        >
          <Text>
            <Text
              style={
                {
                  "color": {
                    "dynamic": {
                      "dark": "#77889d",
                      "highContrastDark": undefined,
                      "highContrastLight": undefined,
                      "light": "#77889d",
                    },
                  },
                }
              }
            >
              HGE2K5VS2R
            </Text>
            <Text
              style={
                {
                  "color": {
                    "dynamic": {
                      "dark": "#172B4D",
                      "highContrastDark": undefined,
                      "highContrastLight": undefined,
                      "light": "#172B4D",
                    },
                  },
                }
              }
            >
               â€¢ 
            </Text>
            <Text
              style={
                {
                  "color": {
                    "dynamic": {
                      "dark": "#77889d",
                      "highContrastDark": undefined,
                      "highContrastLight": undefined,
                      "light": "#77889d",
                    },
                  },
                }
              }
            >
              2024-03-19
            </Text>
          </Text>
        </View>
      </View>
      <View
        style={
          [
            {
              "alignItems": "center",
              "flexDirection": "row",
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": {
                "dynamic": {
                  "dark": "#748094",
                  "highContrastDark": undefined,
                  "highContrastLight": undefined,
                  "light": "#8798ad",
                },
              },
              "fontSize": 14,
              "marginHorizontal": 5,
            }
          }
        >
          4
        </Text>
        <View
          style={
            [
              {
                "overflow": "hidden",
              },
              {
                "height": 22,
                "width": 22,
              },
            ]
          }
        >
          <FastImageView
            accessibilityLabel="fast-image"
            defaultSource={null}
            resizeMode="contain"
            style={
              {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
            testID="fast-image"
          />
        </View>
      </View>
    </View>
    <View
      style={
        [
          {
            "flex": 1,
            "justifyContent": "space-evenly",
          },
        ]
      }
    >
      <View
        style={{}}
      >
        <Text
          ellipsizeMode="tail"
          numberOfLines={2}
          style={
            {
              "color": {
                "dynamic": {
                  "dark": "#172B4D",
                  "highContrastDark": undefined,
                  "highContrastLight": undefined,
                  "light": "#172B4D",
                },
              },
              "flexWrap": "wrap",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Testing 13.02.24
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <Text
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            {
              "color": {
                "dynamic": {
                  "dark": "#77889d",
                  "highContrastDark": undefined,
                  "highContrastLight": undefined,
                  "light": "#77889d",
                },
              },
              "flexShrink": 1,
              "fontSize": 14,
              "fontWeight": "400",
            }
          }
        >
          Argentina Customer
        </Text>
        <Text
          style={
            {
              "color": {
                "dynamic": {
                  "dark": "#77889d",
                  "highContrastDark": undefined,
                  "highContrastLight": undefined,
                  "light": "#77889d",
                },
              },
              "flexShrink": 1,
              "fontSize": 14,
              "fontWeight": "400",
            }
          }
        >
           â€¢ 
        </Text>
        <Text
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            {
              "color": {
                "dynamic": {
                  "dark": "#77889d",
                  "highContrastDark": undefined,
                  "highContrastLight": undefined,
                  "light": "#77889d",
                },
              },
              "flexShrink": 1,
              "flexWrap": "wrap",
              "fontSize": 14,
              "fontWeight": "400",
            }
          }
        >
          Argentina Machine
        </Text>
        <Text
          style={
            {
              "color": {
                "dynamic": {
                  "dark": "#77889d",
                  "highContrastDark": undefined,
                  "highContrastLight": undefined,
                  "light": "#77889d",
                },
              },
              "flexShrink": 1,
              "fontSize": 14,
              "fontWeight": "400",
            }
          }
        >
           â€¢ 
        </Text>
        <Text
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            {
              "color": {
                "dynamic": {
                  "dark": "#77889d",
                  "highContrastDark": undefined,
                  "highContrastLight": undefined,
                  "light": "#77889d",
                },
              },
              "fontSize": 14,
              "fontWeight": "400",
            }
          }
        >
          A-machine
        </Text>
      </View>
    </View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "justifyContent": "space-between",
          },
          {
            "alignItems": "center",
            "flexDirection": "row",
          },
        ]
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              [
                {
                  "overflow": "hidden",
                },
                {
                  "backgroundColor": {
                    "dynamic": {
                      "dark": "#2a2d32",
                      "highContrastDark": undefined,
                      "highContrastLight": undefined,
                      "light": "#E8EAED",
                    },
                  },
                  "borderRadius": 50,
                  "height": 26,
                  "width": 26,
                },
              ]
            }
          >
            <FastImageView
              accessibilityLabel="fast-image"
              defaultSource={null}
              resizeMode="contain"
              style={
                {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
              testID="fast-image"
            />
          </View>
          <Text
            numberOfLines={1}
            style={
              {
                "fontSize": 14,
                "marginHorizontal": 10,
                "textAlign": "center",
              }
            }
          >
            tickets.ticketDetails.notAssigned
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "backgroundColor": "#CCEFEB",
            "borderRadius": 10,
            "padding": 8,
          }
        }
      >
        <Text
          style={
            {
              "color": "#00b09b",
              "fontSize": 13,
            }
          }
        >
          VISIT PLANNED
        </Text>
      </View>
    </View>
  </View>
</View>
`;
